tsconfig.json

```diff
@@ -0,0 +1,18 @@
+{
+  "compilerOptions": {
+    "module": "commonjs",
+    "target": "ES2018",
+    "sourceMap": true,
+    "outDir": "lib",
+    "esModuleInterop": true,
```

Comment:
I would add some strict flags. In the core we have:

```
  "strictNullChecks": true,
  "strictPropertyInitialization": true,
  "strictFunctionTypes": true,
  "strictBindCallApply": true,
  "noImplicitThis": true,
```

I will add that as a separate task